tests/src/ExistingSite/AuthorisationTest.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_showcase\ExistingSite;
+
+/**
+ * Tests that users can only access pages they are authorized to.
+ */
+class AuthorisationTest extends ShowcaseExistingSiteTestBase {
+
+  /**
+   * Anonymous user cannot access restricted pages.
+   */
+  public function testAnonymousAccess(): void {
+    $paths = [
+      '/admin',
+      '/admin/config',
+      '/admin/content',
+      '/admin/people',
+      '/admin/structure',
+      '/node/add',
+    ];
+
+    foreach ($paths as $path) {
+      $this->drupalGet($path);
+      $this->assertSession()->statusCodeEquals(403);
+    }
+  }
+
+  /**
+   * Authenticated user cannot access restricted pages.
+   */
+  public function testAuthenticatedAccess(): void {
+    $paths = [
+      '/admin',
+      '/admin/config',
+      '/admin/content',
+      '/admin/people',
+      '/admin/structure',
+      '/node/add',
+    ];
+
+    foreach ($paths as $path) {
+      $this->drupalGet($path);
+      $this->assertSession()->statusCodeEquals(403);
+    }
+  }
+
+  /**
+   * Manage users user cannot access restricted pages.
+   */
+  public function testManageUsersAccess(): void {
+    $paths = [
+      '/admin/people/roles',
+      '/admin/people/role-settings',
+      '/admin/people/roleassign',
+    ];
+
+    foreach ($paths as $path) {
+      $this->drupalGet($path);
+      $this->assertSession()->statusCodeEquals(403);
+    }
+  }
+
+  /**
+   * Users with the "Manage users" role can assign a limited set of roles.
+   */
+  public function testManageUsersRoleAssign(): void {
+    $user = $this->createUser();
+    $user->addRole('manage_users');
+    $user->save();
+
+    $this->drupalLogin($user);
```

Comment:
We call this regularly, so what about a helper to be called like this:
```suggestion
    $this->loginWithRoles(['manage_users']);
```

Reply:
> One thing I notice: The roles element is disabled in the used edit form display. This means the only way to assign roles is via bulk edit. Is this how we want it to be?

No, we should create a ticket to not hide those elements via form display but via access in form alter